action.yml

```diff
@@ -0,0 +1,82 @@
+name: Configure AWS Profile
+
+description: Configures an OIDC AWS Role Session as an aws config profile.
+
+
+inputs:
+  role-to-assume:
+    description: "ARN of the role to assume."
+    required: true
+  aws-profile-name:
+    description: "Name of profile to be created. If not set ID of step is used."
+    required: false
+  role-duration-seconds:
+    description: "Role duration in seconds. For role chaining max: 3600."
+    default: "7200"
+    required: false
+  aws-region:
+    description: "Name of AWS region to be used."
+    default: "eu-west-1"
+    required: false
+  aws-access-key-id:
+    description: "AWS Access Key ID."
+    required: false
+  aws-secret-access-key:
+    description: "AWS Secret Access Key."
+    required: false
+  aws-session-token:
+    description: "AWS Session Token"
+    required: false
+  aws-source-profile:
+    description: "Name of source profile for assuming role by SDKs"
+    required: false
+
+outputs:
+  aws-profile:
+    description: "Created AWS profile name"
+    value: ${{ inputs.aws-profile-name || github.action }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Profile Based on Source Profile
+      shell: bash
+      if: inputs.aws-source-profile
+      run: |
+        PROFILE_NAME=${{ inputs.aws-profile-name || github.action }}
+        
+        # Set profile for assuming role by SDK
+
+        aws configure set role_arn ${{ inputs.role-to-assume }} --profile ${PROFILE_NAME}
+        aws configure set source_profile ${{ inputs.aws-source-profile }} --profile ${PROFILE_NAME}   
+        aws configure set region ${{ inputs.aws-region }} --profile ${PROFILE_NAME}    
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v2
```

Comment:
all inputs are automatically mask by this action - maybe is better disable masking `mask-aws-account-id: false`